File: JavaScript_Courses/JavaScript_Basics/Preparation/05_While_Loop_Exercise/01_Old_Books.js
function oldBooks(input) {
    let targetBook = input[0];
    let currentIndex = 1;
    let checkedBook = 0;

    while (true) {
        let currentBook = input[currentIndex];

        if (currentBook == "No More Books") {
            console.log("The book you search is not here!");
            console.log(`You checked ${checkedBook} books.`);
            break;
        }

        checkedBook++;
        if (currentBook == targetBook) {
            console.log(`You checked ${checkedBook - 1} books and found it.`);
            break;
        }

        currentIndex++;
    }
}


// oldBooks([
//     "Troy",
//     "Stronger",
//     "Life Style",
//     "Troy"
// ]);

File: Python_Courses/Python_Basics/Python_PB_2025_01/05_While_Loop_Exercise/01_Old_Books.py
target_book = input()  # e.g. 'Troy'

books_checked = 0
is_book_found = False

command = input()  # 'No More Books' or book, e.g. 'Stronger'
while command != 'No More Books':
    current_book = command  # e.g. 'Stronger'

    if current_book == target_book:
        is_book_found = True
        break

    books_checked += 1

    command = input()


if not is_book_found:
    print('The book you search is not here!')
    print(f'You checked {books_checked} books.')
else:
    print(f'You checked {books_checked} books and found it.')
